src/snowflake/cli/plugins/snowpark/package/commands.py

```diff
@@ -131,46 +133,127 @@ def package_upload(
+)
+
+allow_shared_libraries_option = typer.Option(
+    False,
+    "--allow-shared-libraries",
+    help="Allows shared (.so) libraries, when using packages installed through PIP.",
 )
 
 
 @app.command("create", requires_connection=True)
-@cleanup_after_install
+@cleanup_packages_dir
 def package_create(
     name: str = typer.Argument(
         ...,
         help="Name of the package to create.",
     ),
-    install_packages: bool = install_option,
+    ignore_anaconda: bool = ignore_anaconda_option,
+    index_url: Optional[str] = index_option,
+    skip_version_check: bool = skip_version_check_option,
+    allow_shared_libraries: bool = allow_shared_libraries_option,
+    deprecated_allow_native_libraries: PypiOption = deprecated_allow_native_libraries_option,
     _deprecated_install_option: bool = deprecated_install_option,
-    allow_native_libraries: PypiOption = PackageNativeLibrariesOption,
+    _deprecated_install_packages: bool = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Creates a Python package as a zip file that can be uploaded to a stage and imported for a Snowpark Python app.
     """
-    if _deprecated_install_option:
-        install_packages = _deprecated_install_option
+    allow_shared_libraries_pypi_option = {
+        True: PypiOption.YES,
+        False: PypiOption.NO,
+    }[allow_shared_libraries]
+    if deprecated_allow_native_libraries != PypiOption.NO:
+        allow_shared_libraries_pypi_option = deprecated_allow_native_libraries
+
+    if ignore_anaconda:
+        anaconda = None
+    else:
+        anaconda = AnacondaChannel.from_snowflake()
+        package = Requirement.parse(name)
+        if anaconda.is_package_available(
+            package, skip_version_check=skip_version_check
+        ):
+            return MessageResult(
+                f"Package {name} is already available in Snowflake Anaconda Channel."
+            )
 
-    lookup_result = lookup(
-        name=name,
-        install_packages=install_packages,
-        allow_native_libraries=allow_native_libraries,
+    packages_dir = SecurePath(".packages")
+    packages_are_downloaded, dependencies = download_packages(
+        anaconda=anaconda,
+        perform_anaconda_check_for_dependencies=not ignore_anaconda,
+        package_name=name,
+        requirements_file=None,
+        packages_dir=packages_dir,
+        index_url=index_url,
+        allow_shared_libraries=allow_shared_libraries_pypi_option,
+        skip_version_check=skip_version_check,
     )
 
-    if not isinstance(lookup_result, (NotInAnaconda, RequiresPackages)):
-        return MessageResult(lookup_result.message)
+    if not packages_are_downloaded:
+        return MessageResult(
+            dedent(
+                f"""
+                Cannot create package for {name}. Please check the package name
+                or try again with --allow-shared-libraries option.
```

Comment:
If we separate download from checking for shared libraries we avoid `or` messages, we can give users clear information what to do next

Reply:
A separate step is ok too. I just wanted to say that we can have:
```
result:
  - success
  - error1
  - error2
  - errorX
```
And I'm not saying that it's better than separating the logic in this case but I just wanted to show that boolean values are very limited.

Reply:
I've already changed the output to readable class in https://github.com/snowflakedb/snowflake-cli/pull/937